File: Python/Samples/Flyweight/UtFlyweight.py
import unittest
from Models import Content
from ContentFactory import ContentFactory
from CacheFlyweights import CacheFlyweights


class UtFlyweight(unittest.TestCase):

    def test_flyweight(self):
        cacheFw = CacheFlyweights()
        newCache = ContentFactory.CreateProducts()
        cacheFw.add("Products", newCache)

        team = cacheFw.get("Team")
        for item in team:
            print("{0} : {1}".format(item.id, item.value))

        crews = cacheFw.get("Crews")
        for item in crews:
            print("{0} : {1}".format(item.id, item.value))

        prods = cacheFw.get("Products")
        for item in prods:
            print("{0} : {1}".format(item.id, item.value))

        self.assertEqual(len(ContentFactory.CreateTeam()), len(team))
        self.assertEqual(len(ContentFactory.CreateCrews()), len(crews))
        self.assertEqual(len(newCache), len(prods))


if __name__ == '__main__':
    unittest.main()

File: CSharp/DP.UnitTest/UtFlyweight.cs
using System;
using System.Diagnostics;
using DP.Domain.Samples.Flyweight;
using DP.UnitTest.Utility;
using Xunit;
using Xunit.Abstractions;
using System.Collections.Generic;

namespace DP.UnitTest
{
    public class UtFlyweight
    {
        private readonly ITestOutputHelper _output;
        

        public UtFlyweight(ITestOutputHelper output)
        {
            this._output = output;
            System.Diagnostics.Trace.Listeners.Add(new XunitTraceListener(this._output));
        }



        [Fact]
        public void TestFlyweight()
        {
            #region Initialize flyweight objects into CacheFlyweight 
            var cacheFw = new CacheFlyweights();
            var newCache = ContentFactory.CreateProducts();
            cacheFw.Add("Products", newCache);
            #endregion

            var team = cacheFw.Get<List<Content>>("Team");
            team.ForEach( c => Trace.WriteLine($"{c.Id} : {c.Value}"));
            var crews = cacheFw.Get<List<Content>>("Crews");
            crews.ForEach( c => Trace.WriteLine($"{c.Id} : {c.Value}"));            
            var products = cacheFw.Get<List<Content>>("Products");
            products.ForEach( c => Trace.WriteLine($"{c.Id} : {c.Value}"));

            Assert.Equal(team.Count, ContentFactory.CreateTeam().Count);
            Assert.Equal(crews.Count, ContentFactory.CreateCrews().Count);
            Assert.Equal(products.Count, newCache.Count);
            
        }
    }
}
